plugins/permission-common/src/types/permission.ts

```diff
@@ -47,14 +40,45 @@ export type Permission = {
    * all by name.
    */
   attributes: PermissionAttributes;
-  /**
-   * Some permissions can be authorized based on characteristics of a resource
-   * such a catalog entity. For these permissions, the resourceType field
-   * denotes the type of the resource whose resourceRef should be passed when
-   * authorizing.
-   */
-  resourceType?: string;
-};
+} & { type: TType } & TFields;
```

Comment:
Any reason to make `{ type: TType }` an intersection instead of defining the field on `PermissionBase`?

Reply:
Yep! TypeScript does its best to pick a sensible option between expanding type aliases and leaving them in place in type hints and api docs. If the type property is added to `PermissionBase`, it causes TS to expand the type alias. The api-reports in this branch look like this:

```typescript
// @alpha
export const catalogEntityCreatePermission: BasicPermission;
```

But moving `type` into the other object changes them to:

```typescript
// @alpha
export const catalogEntityCreatePermission: {
  name: string;
  attributes: PermissionAttributes;
  type: 'basic';
};
```

To me, the former is way more readable, especially since it makes the different flavors easier to spot when scanning down a list of permissions:

```typescript
// @alpha
export const catalogEntityReadPermission: ResourcePermission<'catalog-entity'>;

// @alpha
export const catalogEntityRefreshPermission: ResourcePermission<'catalog-entity'>;

// @alpha
export const catalogLocationCreatePermission: BasicPermission;

// @alpha
export const catalogLocationDeletePermission: BasicPermission;

// @alpha
export const catalogLocationReadPermission: BasicPermission;
```

Reply:
I noticed here that the `type` property was missing a comment, added in 21162f7d85, and added a comment explaining why the property is separated in 0a76456f58.

Reply:
Ah didn't realize that! I agree the output is more readable with the intersection type. Thanks for adding comments too 💯